modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -94,6 +96,7 @@ private Netty4HttpRequest(
         this.pooled = pooled;
         this.released = released;
         this.inboundException = inboundException;
+        this.rawPath = HttpPreRequest.extractPathFromUri(uri());
```

Comment:
Maybe use netty's [QueryStringDecoder](https://netty.io/4.1/api/io/netty/handler/codec/http/QueryStringDecoder.html) in the implementation?